src/sentry/integrations/utils/code_mapping.py

```diff
@@ -299,36 +301,18 @@ def list_file_matches(self, frame_filename: FrameFilename) -> list[dict[str, str
             ]
 
             for file in matches:
+                stacktrace_root, source_path = find_roots(frame_filename, file)
                 file_matches.append(
                     {
                         "filename": file,
                         "repo_name": repo_tree.repo.name,
                         "repo_branch": repo_tree.repo.branch,
-                        "stacktrace_root": f"{frame_filename.root}/",
-                        "source_path": _get_code_mapping_source_path(file, frame_filename),
+                        "stacktrace_root": stacktrace_root,
+                        "source_path": source_path,
```

Comment:
This function (`list_file_matches`) is used to suggest one or more github repos for a code mapping: 
<img width="608" alt="image" src="https://github.com/getsentry/sentry/assets/55160142/7485adcb-cb23-416a-9ff2-5ed84d99ecfb">
It should use the same, unified logic for deriving the stack trace root and source code root.